cylc/flow/data_store_mgr.py

```diff
@@ -1972,10 +1972,13 @@ def prune_pruned_updated_nodes(self):
                     node = tp_added[tp_id]
                 else:
                     continue
-                for j_id in list(node.jobs) + list(tp_updated[tp_id].jobs):
+                update_node = tp_updated[tp_id]
```

Comment:
...but can pop instead
```suggestion
                update_node = tp_updated.pop(tp_id)
```